spec/unit/shoulda/matchers/active_record/validate_uniqueness_of_matcher_spec.rb

```diff
@@ -283,6 +285,27 @@
         column_type: :string
     end
 
+    context 'when one of the scoped attributes is a boolean column' do
+      include_context 'it supports scoped attributes of a certain type',
+        column_type: :boolean
+
+      context "when more than one record exists that has the next version of the attribute's value" do
+        it 'rejects' do
+          value1 = dummy_value_for(:boolean)
+          value2 = next_version_of(value1, :boolean)
+          value3 = next_version_of(value2, :boolean)
+          model = define_model_validating_uniqueness(
+            scopes: [ build_attribute(name: :scope) ]
+          )
+          create_record_from(model, scope: value2)
+          create_record_from(model, scope: value3)
+          record = build_record_from(model, scope: value1)
+
+          expect(record).not_to validate_uniqueness.scoped_to(:scope)
```

Comment:
This should actually accept, I think...

So... this is something I actually noticed a while back. We're actually missing a check here. If you're validating uniqueness of a given attribute and it's scoped, then the first thing you should check is, if I change the value of that attribute to a unique value, then it should make the record valid. But currently we just loop through all the scope attributes and assert that if we change those to unique values then it should make the record valid.

But even if we added this check then the matcher would still fail. This is because we are still trying to check that column within #validate_after_scope_change?. We need to somehow exclude the check in this case, since both true and false are taken as possible next values.